gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -1059,13 +1064,50 @@ private static Map<String, FileDescriptor> getFilesToGenerate(CodeGeneratorReque
 
       fileDescriptors.put(fileDescriptor.getName(), fileDescriptor);
     }
+    extraMixins.forEach(fileDescriptor -> parseDependencies(fileDescriptor, fileDescriptors));
```

Comment:
I _think_ we can just add the three mixin FileDescriptor to `fileDescriptors` here. Can you please give it a try?

Reply:
done

Reply:
It seems that if we always add Mixin protos, in `reflect-config.json`, all message types associated with Mixins are generated (difference: https://paste.googleplex.com/6265925991923712).

Is this OK?

Reply:
If we *always* add all Mixins, then `reflect-config.json` will contain message types associated with all Mixins. Previously, we only add Mixins defined in service yaml so certain message types are excluded.

I changed the logic to only add fileDescriptor of Mixins, but not fileDescriptors of their dependencies, but the result is the same (all message types associated with Mixins are in `reflect-config.json`).

> Ideally, we can keep all other logics and this is all we need to change in this PR

We need to change the logic of getting filesToGenerate as well; otherwise Mixin methods won't be generated. I confirmed this with unit tests.
https://github.com/googleapis/sdk-platform-java/blob/1b861ec78a7d48549140224cc15aa793e3d890ec/gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java#L245

Reply:
> We need to change the logic of getting filesToGenerate as well; otherwise Mixin methods won't be generated. I confirmed this with unit tests.

This part is gone.